canonical-web-and-design/maas-ui
# fix(ui): prevent long tag names from breaking layout

## Done

- Added a css util `.u-break-word` and used in a few places that tags might show in order to prevent long tag names from breaking the layout.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine list, select a machine and add some really long tag names
- Check that the tag select field doesn't look completely broken
- Open the details page of the tagged machine and check that the summary page isn't broken at any screen size
- Check the same on the configuration page

## Fixes

Fixes #3198 

## Screenshot
![2021-11-10_16-16](https://user-images.githubusercontent.com/25733845/141060187-a7758bb5-c145-4a8e-a01b-26ea884e5de4.png)
